### PR TITLE
GPG article

### PR DESCRIPTION
**Pull request by HarithaIBM**

### CODE REVIEW

File: docs/Guides/GpgOnZOS.md [hunk 1]
```diff
@@ -0,0 +1,103 @@
+# **GPG ON ZOS**
```
Comment (IgorTodorovskiIBM) [inline docs/Guides/GpgOnZOS.md:1]:
```suggestion
# **GPG on z/OS**
```

File: docs/Guides/GpgOnZOS.md [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+
+
+### What is Gpg?
```
Comment (IgorTodorovskiIBM) [inline docs/Guides/GpgOnZOS.md:4]:
```suggestion
### What is GPG?
```

File: docs/Guides/GpgOnZOS.md (hunk 1 above)
Comment (HarithaIBM) [reply docs/Guides/GpgOnZOS.md:1]: Corrected it.

File: docs/Guides/GpgOnZOS.md (hunk 2 above)
Comment (HarithaIBM) [reply docs/Guides/GpgOnZOS.md:4]: Corrected it.

File: docs/Guides/GpgOnZOS.md [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+
+GPG allows users to generate key pairs consisting of a public key and a private key. The public key can be freely distributed, while the private key must be kept secret. With GPG, users can encrypt data using the recipient's public key, ensuring that only the intended recipient can decrypt and access the information using their private key.
+
+GPG also supports digital signatures, which provide a way to verify the authenticity and integrity of data. By signing a document with their private key, a user can demonstrate that the document has not been tampered with and was indeed created by them. The signature can be verified using the corresponding public key.
+
+Overall, GPG is commonly used for secure communication, email encryption, file encryption, and verifying the authenticity of software distributions. It is widely regarded as a reliable and trusted tool for ensuring privacy and security in digital communications.
+
+
+
+### How to generate GPG key?
```
[the remark this answers is not in the document]
Comment (HarithaIBM) [reply docs/Guides/GpgOnZOS.md:18]: Corrected it. Thank you

**Review by MikeFultonDev — APPROVED** [review]

File: docs/Guides/GpgOnZOS.md [hunk 4]
```diff
@@ -0,0 +1,103 @@
...
+### How to Sign Git commits using GPG key?
+
+1. Add your public GPG Key to your account on github.com and refer the [link](https://docs.github.com/en/authentication/managing-commit-signature-verification/adding-a-gpg-key-to-your-github-account) for steps.
+
+2. Tell git about your public key which will be used for signing. You can get steps from [here](https://docs.github.com/en/authentication/managing-commit-signature-verification/telling-git-about-your-signing-key#telling-git-about-your-gpg-key)
+
+3. Now you can sign your commit using command (provide your phrase on prompt)
+	`git commit -S -m "YOUR_COMMIT_MESSAGE"`
+	
+1. 	When you've finished creating commits locally, push them to your remote repository on GitHub:
```
Comment (IgorTodorovskiIBM) [inline docs/Guides/GpgOnZOS.md:62]: Should this be 4.? Rather than 1.

File: docs/Guides/GpgOnZOS.md [hunk 5]
```diff
@@ -0,0 +1,103 @@
...
+
+	`git push `
+1. 	On GitHub, navigate to your pull requestO and on the pull request, click  Commits. To view more detailed information about the verified signature, click `Verified`.
+
+
+### How to sign images?
+1. Any file or image can be signed using command below.
+	`gpg --output <image.name>.sig --sign <image-name>`
+	
+1. 	Signed file/image can be verified using:
```
Comment (IgorTodorovskiIBM) [inline docs/Guides/GpgOnZOS.md:72]: Should this be 2?

File: docs/Guides/GpgOnZOS.md [hunk 6]
```diff
@@ -0,0 +1,103 @@
...
+	
+1. 	Both image/file and the detached signature are needed to verify the signature using --verify option:
+	`gpg --verify <image.name>.sig <image.name>`
+	
+### Example:
+
+1. Download signed bash tool in tar.gz format using command shown below:
+	`curl -k -O http://ftp.vim.org/ftp/ftp/gnu/bash/bash-5.2.15.tar.gz.sig`
+	
+1. 	Download the keyring for the same, using command shown below:
```
Comment (IgorTodorovskiIBM) [inline docs/Guides/GpgOnZOS.md:92]:
```suggestion
1. 	Download the public GNU keyring using command shown below:
```

File: docs/Guides/GpgOnZOS.md [hunk 7]
```diff
@@ -0,0 +1,103 @@
...
+	`curl -k -O https://ftp.gnu.org/gnu/gnu-keyring.gpg` 
+	
+1. 	Verify the downloaded tar file , using command:
+	`gpg --verify --keyring ./gnu-keyring.gpg bash-5.2.15.tar.gz.sig`
```
Comment (IgorTodorovskiIBM) [inline docs/Guides/GpgOnZOS.md:96]: Can you paste the expected output as well

File: docs/Guides/GpgOnZOS.md [hunk 8]
```diff
@@ -0,0 +1,103 @@
...
+	`gpg --output <image.name>.sig --sign <image-name>`
+	
+1. 	Signed file/image can be verified using:
+	`gpg --verify <image.name>.sig`
+	
+1. 	To verify the signature and extract the file/image use the --decrypt option. The signed file/image to verify and recover is input and the recovered document is output.
+	`gpg --output zoslib-zopen.20230511_114620.zos.pax.Z --decrypt zoslib-zopen.20230511_114620.zos.pax.Z.gpg`
+	
+
+### How to	create detached signature and sign an image?
```
Comment (IgorTodorovskiIBM) [inline docs/Guides/GpgOnZOS.md:79]:
```suggestion
### How to create detached signature and sign an image?
```

File: docs/Guides/GpgOnZOS.md [hunk 9]
```diff
@@ -0,0 +1,103 @@
...
+
+1. A detahced signature can be created and an image/file can signed using the follwing commad:
```
Comment (IgorTodorovskiIBM) [inline docs/Guides/GpgOnZOS.md:81]:
```suggestion
1. A detached signature can be created and an image/file can signed using the follwing commad:
```

**Comment by IgorTodorovskiIBM** [comment]
A few general comments:
* Add link to the official gpg site in case users want to find out more information
* Add a link to the gpgport release or describe how to download it using zopen

Otherwise, looks great!

File: docs/Guides/GpgOnZOS.md (hunk 4 above)
Comment (HarithaIBM) [reply docs/Guides/GpgOnZOS.md]: Corrected it

File: docs/Guides/GpgOnZOS.md (hunk 5 above)
Comment (HarithaIBM) [reply docs/Guides/GpgOnZOS.md]: Corrected it

File: docs/Guides/GpgOnZOS.md (hunk 6 above)
Comment (HarithaIBM) [reply docs/Guides/GpgOnZOS.md]: Corrected it

File: docs/Guides/GpgOnZOS.md (hunk 7 above)
Comment (HarithaIBM) [reply docs/Guides/GpgOnZOS.md:109]: Done

**Review by IgorTodorovskiIBM — APPROVED** [review]
LGTM!

**Pull request merged**